images/hubble-proto/install-protoplugins.sh

```diff
@@ -14,12 +14,13 @@ go mod init github.com/cilium/hubble/protoc
 
 # latest tag at the time. For some reason `go get foo/bar/baz@vX.Y.Z` doesn't
 # work with nested go.mod definitions.
-go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@cee815d
+# a80e650 == tag: cmd/protoc-gen-go-grpc/v1.0.1
+go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@a80e65018272ac6bd58ee44d0d3e37374b090c1f
```

Comment:
Cannot comment on the lines above, but dropping the `unset GOPATH` and `go mod init` and instead setting `GO111MODULE=on` (which will be the default in the upcoming Go 1.16) and `GOPATH=/proto` would allow to use single line `go get ...` with proper version tags to get and build all the binaries. However, they will be installed to `$GOPATH/bin` rather than `/proto` (haven't found a way to override this behavior), so the `COPY` in `Dockerfile` would need to be adjusted as well.

Something like the following:

```bash
#!/bin/bash

set -o xtrace
set -o errexit
set -o pipefail
set -o nounset

mkdir /proto

export GOPATH=/proto
export GO111MODULE=on

go get google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.0.1

# protoc-gen-go-json doesn't have releases, this is the latest commit at the time
go get github.com/mitchellh/protoc-gen-go-json@364b693

go get github.com/golang/protobuf@v1.4.3
```

and the diff for `Dockerfile`:

```diff
--- a/images/hubble-proto/Dockerfile
+++ b/images/hubble-proto/Dockerfile
@@ -33,9 +33,9 @@ RUN /tmp/install-glibc.sh
 COPY install-protoc.sh /tmp/install-protoc.sh
 RUN /tmp/install-protoc.sh
 
-COPY --from=builder /proto/protoc-gen-go .
-COPY --from=builder /proto/protoc-gen-go-grpc .
-COPY --from=builder /proto/protoc-gen-go-json .
+COPY --from=builder /proto/bin/protoc-gen-go .
+COPY --from=builder /proto/bin/protoc-gen-go-grpc .
+COPY --from=builder /proto/bin/protoc-gen-go-json .
```

Reply:
Thanks. Just tried it but it's somehow not working (I didn't investigate further, not sure it's worth the trouble).

Reply:
Certainly not worth the trouble and if you want I can try to take this up in a follow-up PR once this one is merged.